src/main.rs

```diff
@@ -69,6 +74,17 @@ fn main() -> ! {
     wdt1.disable();
     {% endif -%}
 
+    {% if logging -%}
```

Comment:
I think using `init_logger_from_env()` is easier - plus it supports `ESP_LOGTARGET`.
Difference is it defaults to `Off` not `Info`

Reply:
True.
This is what i used in my code because i wanted logging in the default mode.
But I think changing it and adding a comment that you have to set ESP_LOGLEVEL to use it should be fine

Reply:
Do you have an idea if there is any way to set the ENV variable somewhere in the cargo config so that `cargo run` just works and prints logs?

Reply:
Yes, see here: https://doc.rust-lang.org/cargo/reference/config.html#configuration-format

Reply:
sadly all of that only works for your own crate, not for dependencies (in this case esp_println)
At least i couldn't figure out a way to get log output, but printing ESP_LOGLEVEL from the config in my own code worked.
So it looks like the rustc-env is only applied to your own code and the ENV section only at runtime

Reply:
No, it works for dependencies too. Smoltcp does its configuration this way. There is a horrible bug in cargo which means it doesn't detect changes with the `[env]` section: https://github.com/rust-lang/cargo/issues/10358, so make sure whenever you add/change/remove the `[env]` section do a clean build afterwards.

Reply:
Ah ok
Thanks for the information.